velox/common/memory/MappedMemory.cpp

```diff
@@ -338,6 +338,64 @@ std::shared_ptr<MappedMemory> MappedMemory::addChild(
   return std::make_shared<ScopedMappedMemory>(this, tracker);
 }
 
+namespace {
+// Returns the size class size that corresponds to 'size'.
+int32_t sizeClassSize(
+    int32_t size,
+    const std::vector<MachinePageCount>& sizes) {
+  VELOX_CHECK_LE(size, sizes.back());
+  return *std::lower_bound(sizes.begin(), sizes.end(), size);
+}
+} // namespace
+
+void* FOLLY_NULLABLE
+MappedMemory::allocateBytes(uint64_t size, int32_t maxMallocSize) {
+  if (size <= maxMallocSize) {
+    return reinterpret_cast<char*>(::malloc(size));
+  } else if (size <= sizeClassSizes_.back() * kPageSize) {
+    Allocation allocation(this);
+    auto numPages = sizeClassSize(
+        bits::roundUp(size, kPageSize) / kPageSize, sizeClassSizes_);
```

Comment:
We can move the bits::roundUp logic into the helper method as we call this roundup every time, so that we can directly pass the byte size to the method and get size class size out from it.